libraries/abstractions/pkcs11/psa/iot_pkcs11_psa.c

```diff
@@ -0,0 +1,2337 @@
+	{
+		if( ( NULL == pxTemplate ) || ( 0 == ulCount ) )
+		{
+			xResult = CKR_ARGUMENTS_BAD;
+		}
+	}
+	if( xResult == CKR_OK )
+	{
+		/*
+		 * Find the object in P11context firstly.
+		 */
+		prvFindObjectInListByHandle( xObject, &xPalHandle, &pcLabel, &xSize ); /*pcLabel and xSize are ignored. */
+
+		if( xPalHandle != CK_INVALID_HANDLE )
+		{
+			xResult = PKCS11_PSA_GetObjectValue( xPalHandle, ucGetData, &ulLength, &xIsPrivate );
+			if( xResult == CKR_OK )
+			{
+				pxObjectValue = ucGetData;
+			}
+		}
+		else
+		{
+			xResult = CKR_OBJECT_HANDLE_INVALID;
+		}
+	}
+
+	if( xResult == CKR_OK )
+	{
+		for( iAttrib = 0; iAttrib < ulCount && CKR_OK == xResult; iAttrib++ )
+		{
+			switch( pxTemplate[ iAttrib ].type )
+			{
+				case CKA_CLASS:
+
+					if( pxTemplate[ iAttrib ].pValue == NULL )
+					{
+						pxTemplate[ iAttrib ].ulValueLen = sizeof( CK_OBJECT_CLASS );
+					}
+					else
+					{
+						if( pxTemplate[ iAttrib ].ulValueLen >= sizeof( CK_OBJECT_CLASS ) )
+						{
+							switch (( P11ObjectHandles_t )xPalHandle )
+							{
+								case eAwsDevicePrivateKey:
+									xClass = CKO_PRIVATE_KEY;
+									break;
+
+								case eAwsDevicePublicKey:
+								case eAwsCodeSigningKey:
+									xClass = CKO_PUBLIC_KEY;
+									break;
+
+								case eAwsJitpCertificate:
+								case eAwsRootCertificate:
+								case eAwsDeviceCertificate:
+									xClass = CKO_CERTIFICATE;
+									break;
+
+								default:
+									xResult = CKR_OBJECT_HANDLE_INVALID;
+									break;
+							}
+							memcpy( pxTemplate[ iAttrib ].pValue, &xClass, sizeof( CK_OBJECT_CLASS ) );
+						}
+						else
+						{
+							xResult = CKR_BUFFER_TOO_SMALL;
+						}
+					}
+
+					break;
+
+				case CKA_VALUE:
+
+					if( xIsPrivate == CK_TRUE )
+					{
+						pxTemplate[ iAttrib ].ulValueLen = CK_UNAVAILABLE_INFORMATION;
+						xResult = CKR_ATTRIBUTE_SENSITIVE;
+					}
+					else
+					{
+						if( pxTemplate[ iAttrib ].pValue == NULL )
+						{
+							pxTemplate[ iAttrib ].ulValueLen = ulLength;
+						}
+						else if( pxTemplate[ iAttrib ].ulValueLen < ulLength )
+						{
+							xResult = CKR_BUFFER_TOO_SMALL;
+						}
+						else
+						{
+							memcpy( pxTemplate[ iAttrib ].pValue, pxObjectValue, ulLength );
+						}
+					}
+
+					break;
+
+				case CKA_KEY_TYPE:
+
+					if( pxTemplate[ iAttrib ].pValue == NULL )
+					{
+						pxTemplate[ iAttrib ].ulValueLen = sizeof( CK_KEY_TYPE );
+					}
+					else if( pxTemplate[ iAttrib ].ulValueLen < sizeof( CK_KEY_TYPE ) )
+					{
+						xResult = CKR_BUFFER_TOO_SMALL;
+					}
+					else
+					{
+						if( 0 != xResult )
+						{
+							xResult = CKR_FUNCTION_FAILED;
+						}
+						else
+						{
+							psa_key_handle_t ulKeyHandle = 0;
+							if( xPalHandle == eAwsDevicePrivateKey )
+							{
+								ulKeyHandle = P11KeyConfig.uxDevicePublicKey;
+							}
+							else if( xPalHandle == eAwsDevicePublicKey )
+							{
+								ulKeyHandle = P11KeyConfig.uxDevicePrivateKey;
```

Comment:
should be ulKeyHandle = P11KeyConfig.uxDevicePublicKey;

Reply:
yes, I will fix it in the next commit.